dask/highgraph.py

```diff
@@ -0,0 +1,55 @@
+from . import sharedict
+from .utils import ignoring
+from .base import is_dask_collection
+
+import toolz
+
+class HighGraph(sharedict.ShareDict):
```

Comment:
Two thoughts:
1. I like the name `HighLevelGraph` better than `HighGraph`.
2. Can you make this extend ShareDict with composition instead of inheritance? I find that easier to understand and less error prone.

Reply:
> I would like to just replace ShareDict entirely.

OK, works for me. I just wouldn't bother with inheritance, then.

Reply:
Agreed, its only there short term as I trade things out